stdlib/jdbc/src/main/java/org/ballerinax/jdbc/statement/BatchUpdateStatement.java

```diff
@@ -83,8 +87,8 @@ public BatchUpdateStatement(ObjectValue client, SQLDatasource datasource, String
         String errorMessagePrefix = "failed to execute batch update";
         try {
             conn = getDatabaseConnection(strand, client, datasource);
-            boolean generatedKeyReturningSupported = isGeneratedKeyReturningSupported();
-            if (generatedKeyReturningSupported) {
+            boolean keyRetrievalSupportedStatement = datasource.isKeyRetrievalSupported();
+            if (getGeneratedKey && keyRetrievalSupportedStatement) {
```

Comment:
Shall we remove checking `keyRetrievalSupportedStatement` and only rely on value set by the user? 
and pass all the errors we encounter while performing the operation to the user.